builder/virtualbox/common/driver_4_2.go

```diff
@@ -76,14 +76,7 @@ func (d *VBox42Driver) CreateSCSIController(vmName string, name string) error {
 }
 
 func (d *VBox42Driver) Delete(name string) error {
-	ctx := context.TODO()
-	return retry.Config{
-		Tries:      5,
-		RetryDelay: (&retry.Backoff{InitialBackoff: 1 * time.Second, MaxBackoff: 1 * time.Second, Multiplier: 2}).Linear,
-	}.Run(ctx, func(ctx context.Context) error {
-		err := d.VBoxManage("unregistervm", name, "--delete")
-		return err
-	})
```

Comment:
Removing code is always nice 🙂.

This code retries for any case; but the code in `VBoxManage` only retries when `strings.Contains(err.Error(), "VBOX_E_INVALID_OBJECT_STATE")` is true. Is that okay ?

Reply:
I think it is because it was added for fixing the lock session thing, and the retry inside VBoxManage is there for the same reason but is more explicit about the reasoning around retrying.  I think it's ok, do you? 🤔

Reply:
That sounds good to me; my concern is that there's probably going to be that weird case we missed. But it won't be too bad and the issue will bubble up someday.